packages/klighd-core/src/bookmarks/bookmark-panel.tsx

```diff
@@ -0,0 +1,228 @@
+                >
+                    <i attrs={{ "data-feather": "bookmark" }} />
+                </button>
+                <button
+                    title="Load from Clipboard"
+                    classNames="options__icon-button"
+                    on-click={() => this.handleBookmarkPast()}
+                >
+                    <i attrs={{ "data-feather": "upload" }} />
+                </button>
+            </div>
+            {this.renderBookmarkList()}
+        </div>;
+    }
+
+    private renderBookmarkList(): VNode {
+        return <div>
+            {
+                this.bookmarkRegistry.bookmarks.map((bookmark) => (
+                    this.renderBookmarkEntry(bookmark)
+                ))
+            }
+        </div>
+    }
+
+    private renderBookmarkEntry(bookmark: Bookmark): VNode {
+        return <div key={bookmark.bookmarkIndex}>
+            <fieldset>
+                <legend>{bookmark.name ?? ("Bookmark " + bookmark.bookmarkIndex)}</legend>
+                <button
+                    title="Goto"
+                    classNames="options__icon-button"
+                    on-click={() => this.handleBookmarkGoto(bookmark)}
+                >
+                    <i attrs={{ "data-feather": "map-pin" }} />
+                </button>
+                <button
+                    title="Save to Clipboard"
+                    classNames="options__icon-button"
+                    on-click={() => this.handleBookmarkCopy(bookmark)}
+                >
+                    <i attrs={{ "data-feather": "download" }} />
+                </button>
+                <button
+                    title="Delete Bookmark"
+                    classNames="options__icon-button"
+                    on-click={() => this.deleteBookmark(bookmark)}
+                >
+                    <i attrs={{ "data-feather": "trash-2" }} />
+                </button>
+                <button
+                    id={bookmark.editId}
+                    title="Edit Bookmark Name"
+                    classNames="options__icon-button"
+                    on-click={() => this.startBookmarkNameEdit(bookmark)}
+                >
+                    <i attrs={{ "data-feather": "edit-3" }} />
+                </button>
+                <span id={bookmark.saveId} classNames="options__hidden">
+                    <button
+                        title="Save Bookmark Name"
+                        classNames="options__icon-button"
+                        on-click={() => this.saveBookmarkName(bookmark)}
+                    >
+                        <i attrs={{ "data-feather": "check" }} />
+                    </button>
+                    <input type="text" value={bookmark.name ?? ""} placeholder="Enter Bookmark Name" on-keypress={(event: KeyboardEvent) => this.inputSaveBookmarkName(bookmark, event)} />
+                </span>
+            </fieldset>
+        </div>
+    }
+
+    private handleBookmarkGoto(bookmark: Bookmark) {
```

Comment:
You configured the registry to be configurable by injection, for that to work properly the methods should all be protected here, so that someone extending or altering the behavior can do so in an easy fashion.